src/blas_like/level1/MaxAbs.cpp

```diff
@@ -11,46 +11,58 @@
 
 namespace El {
 
-    template<typename Real,typename>
-    Real MaxAbs( const Matrix<Real>& A )
+template<typename Real>
+Base<Real> MaxAbs( const Matrix<Real>& A )
+{
+    DEBUG_CSE
+    const Int m = A.Height();
+    const Int n = A.Width();
+    const Real* ABuf = A.LockedBuffer();
+    const Int ALDim = A.LDim();
+
+    Real value = 0;
+    for( Int j=0; j<n; ++j )
+        for( Int i=0; i<m; ++i )
+            value = Max(abs(value),ABuf[i+j*ALDim]);
```

Comment:
The `std::abs` function only works with standard C++ datatypes, such as `int`, `float`, `std::complex<float>`, and `std::complex<double>`. But they don't work for Elemental's plethora of other datatypes, such as `El::DoubleDouble`, `El::QuadDouble`, `El::Quad, `El::BigFloat`, `El::Complex<El::DoubleDouble>`, `El::Complex<El::QuadDouble>`, `El::Complex<El::Quad>`, or `El::Complex<El::BigFloat>`. For this reason it is necessary to use `El::Abs`. I am assuming that you built Elemental without support for QD, __float128, or MPFR.